src/runtime_src/doc/toc/yocto.rst

```diff
@@ -22,9 +18,61 @@ Before start to build Linux image, make sure your have:
 The PetaLinux tool chain can be downloaded from xilinx.com.
 If you don't have a .hdf file, please see :ref:`Build Boot Images`.
 
+*NOTE* The purpose of this page is only for providing general getting started guidance.
+If you are interested in details about creating software images for embedded platform.
+Please read the Software Platform section of Xilinx® Document UG1146.
+
 Create PetaLinux Project with XRT recipes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+**NOTE**: Beginning in Xilinx release 2019.2, PetaLinux contains recipes for XRT
+and all required software prerequisites. This information is provided as reference
+for those who wish to build within a standard Yocto flow *or* for those who wish
+to customize or extend these recipes. For most use cases, *do not* add the recipes
+from the XRT repository into a 2019.2+ PetaLinux project and instead enable
+XRT through the root file system menuconfig utility.
+
+.. code-block:: bash
+
+        #Configure Linux kernel (default kernel config is good for zocl driver)
+        $ petalinux-config -c kernel
+
+        # Configure rootfs, enable below components:
+        #   menu -> "Filesystem Pacakges" -> "libs" -> xrt
+        #   menu -> "Filesystem Pacakges" -> "libs" -> xrt-dev
+        $ petalinux-config -c rootfs
+
+Add XRT kernel node in device tree
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+An example of device tree is in ``<XRT>/src/runtime_src/core/edge/fragments/xlnk_dts_fragment_mpsoc.dts``. You can attach it to ``project-spec/meta-user/recipes-bsp/device-tree/files/system-user.dtsi`` in your PetaLinux project.
+
+Building the PetaLinux kernel image
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: bash
+
+        # Build package
+        $ petalinux-build
+
+After building you will find all output files from images/linux directory in your PetaLinux project.
+These files can be used when creating an embedded platform.
+
+- ``image.ub``
+- ``bl31.elf``
+- ``fsbl.elf``
+- ``pmufw.elf``
+- ``u-boot.elf``
+
+
+Using custom recipes
+~~~~~~~~~~~~~~~~~~~~
+
+**NOTE: For most use cases these steps are NOT REQUIRED. Most users will not need the information
+below this point; it is provided for reference only for those who wish to customize/extend the build
+system or use these recipes with Yocto builds.**
+
+
```

Comment:
I think we should just delete anything below this line.
To customize XRT recipes, user dones't need new bb files. Yocto system provide standard way, .bbappend file, to customize a recipe. We could said this section required knowledge on Yocto.

Steps:
1. Create a petalinux project
2. Create and go to project-spec/meta-user/recipes-xrt
3. Create directory and file, xrt/xrt_%.bbappend
4. You could change anything you want. Ex. SRCURI, commit number, branch and even use EXTERNALSRC class.

The xrt bb file could be found at https://github.com/Xilinx/meta-xilinx/blob/master/meta-xilinx-bsp/recipes-xrt/xrt/xrt_git.bb
The zocl bb file could be found at https://github.com/Xilinx/meta-xilinx/blob/master/meta-xilinx-bsp/recipes-xrt/zocl/zocl_git.bb